service/internal/proctelemetry/process_telemetry.go

```diff
@@ -127,6 +166,96 @@ func RegisterProcessMetrics(registry *metric.Registry, ballastSizeBytes uint64)
 	return nil
 }
 
+func (pm *processMetrics) recordWithOtel(ctx context.Context) error {
+	var err error
+
+	pm.otelProcessUptime, err = pm.meter.AsyncFloat64().Counter(
+		"process_uptime",
+		instrument.WithDescription("Uptime of the process"),
+		instrument.WithUnit(unit.Unit("s")))
+	if err != nil {
+		return err
+	}
+	err = pm.meter.RegisterCallback([]instrument.Asynchronous{pm.otelProcessUptime}, func(ctx context.Context) {
+		pm.otelProcessUptime.Observe(ctx, pm.updateProcessUptime())
+	})
+	if err != nil {
+		return err
+	}
+
+	pm.otelAllocMem, err = pm.meter.AsyncInt64().Gauge(
+		"process_runtime_heap_alloc_bytes",
+		instrument.WithDescription("Bytes of allocated heap objects (see 'go doc runtime.MemStats.HeapAlloc')"),
+		instrument.WithUnit(unit.Bytes))
+	if err != nil {
+		return err
+	}
+	err = pm.meter.RegisterCallback([]instrument.Asynchronous{pm.otelAllocMem}, func(ctx context.Context) {
+		pm.otelAllocMem.Observe(ctx, pm.updateAllocMem())
+	})
+	if err != nil {
+		return err
+	}
+
+	pm.otelTotalAllocMem, err = pm.meter.AsyncInt64().Counter(
+		"process_runtime_total_alloc_bytes",
+		instrument.WithDescription("Cumulative bytes allocated for heap objects (see 'go doc runtime.MemStats.TotalAlloc')"),
+		instrument.WithUnit(unit.Bytes))
+	if err != nil {
+		return err
+	}
+	err = pm.meter.RegisterCallback([]instrument.Asynchronous{pm.otelTotalAllocMem}, func(ctx context.Context) {
+		pm.otelTotalAllocMem.Observe(ctx, pm.updateTotalAllocMem())
+	})
+	if err != nil {
+		return err
+	}
+
+	pm.otelSysMem, err = pm.meter.AsyncInt64().Gauge(
+		"process_runtime_total_sys_memory_bytes",
+		instrument.WithDescription("Total bytes of memory obtained from the OS (see 'go doc runtime.MemStats.Sys')"),
+		instrument.WithUnit(unit.Bytes))
+	if err != nil {
+		return err
+	}
+	err = pm.meter.RegisterCallback([]instrument.Asynchronous{pm.otelSysMem}, func(ctx context.Context) {
+		pm.otelSysMem.Observe(ctx, pm.updateSysMem())
+	})
+	if err != nil {
+		return err
+	}
+
+	pm.otelCpuSeconds, err = pm.meter.AsyncFloat64().Counter(
+		"process_cpu_seconds",
+		instrument.WithDescription("Total CPU user and system time in seconds"),
+		instrument.WithUnit(unit.Unit("s")))
+	if err != nil {
+		return err
+	}
+	err = pm.meter.RegisterCallback([]instrument.Asynchronous{pm.otelCpuSeconds}, func(ctx context.Context) {
+		pm.otelCpuSeconds.Observe(ctx, pm.updateCPUSeconds())
+	})
+	if err != nil {
+		return err
+	}
+
+	pm.otelRssMemory, err = pm.meter.AsyncInt64().Gauge(
```

Comment:
nit: abbreviations should be all Uppercase if it is not in the begin of the variable name
```suggestion
	pm.otelRSSMemory, err = pm.meter.AsyncInt64().Gauge(
```